experiment (#41)
SpecialeBajs/Rapport: article/experiment.tex

\section{Experiment}
\input{article/equal-data.tex}

\section{Simple price aware extension of LightGCN}
To investigate the extension of LightGCN, we investigate the performance of Price-Aware Recommendation, LightGCN, GCN, NGCF, GCMC, adapted LightGCN with Price and category nodes in the input data and NGCF with a price and category extension.

> The hyperparameter $X$ is tested with the values of $0.0$, $0.5$, $1.0$ and $2.0$ to see the effect of this as input data has on the final result.

Reviewer: beskriver vi tidligere hvad X gør? aka en vægt for cat og pris

Author: Vi skriver "If there is a connection in $R_{ic}$ or $R_{ip}$ this value is a hyperparameter $X$ with a value $x>0$, otherwise it is 0." i implementationsafsnittet. Ved ikke om det er nok.

        LightGCN       & \textbf{0.2106}                                           & \textbf{0.1063} & \textbf{0.3176} & \textbf{0.1344} \\ \hline
        PUP            & 0.1697                                                    & 0.07802         & 0.2654          & 0.1023          \\ \hline
        GCN            & 0.1558                                                    & 0.07593         & 0.2442          & 0.1001          \\ \hline
        NGCF           & 0.1810                                                    & 0.08817         & 0.2769          & 0.1132          \\ \hline
        GCMC           & 0.1692                                                    & 0.0835          & 0.2497          & 0.1008          \\ \hline
        PAS (1.0)      & 0.1542                                                    & 0.0717          & 0.2199          & 0.086           \\ \hline
        NGCF PAS (1.0) & 0.1749                                                    & 0.0849          & 0.2743          & 0.1111          \\ \hline
    \end{tabular}
    \caption{Results for the experiment with the different methods}
    \label{tab:results-with-many-methods}
\end{table*}

> \subsubsection{Hyperparameter experiment}

Author: Læs det her afsnit

As described in \autoref{subsec:simple-extension}, we utilize a hyperparameter that is inserted into the adjacency matrix, whenever there is a connection between the item nodes and category nodes, or item nodes and price nodes.

> The hyperparameter $X$ in LGCN PAS and NGCF PAS is tested with the values of $0.0$, $0.5$, $1.0$ and $2.0$ to see the effect of this as input data has on the final result.

Reviewer 2: ```suggestion
Experiments are done with the hyperparameter $X$ in LGCN PAS and NGCF PAS with the values $0.0$, $0.5$, $1.0$ and $2.0$ to see what effect of this has on the performance.
```

\begin{table*}[h!]
    \centering
    \begin{tabular}{|l|l|l|l|l|}
        \hline
        \rowcolor[HTML]{FFFFFF}
                       & \multicolumn{4}{l|}{\cellcolor[HTML]{FFFFFF}Yelp Dataset}                                   \\ \hline
        Method         & Recall@50                                                 & NDCG@50 & Recall@100 & NDCG@100 \\ \hline

>         PAS (0.0)      & 0.1560                                                    & 0.07674 & 0.2456     & 0.09901  \\ \hline

Reviewer: lgcn pas

        LGCN PAS (0.5) & 0.1591                                                    & 0.07825 & 0.2539     & 0.1010   \\ \hline
        LGCN PAS (1.0) & 0.1542                                                    & 0.0717  & 0.2199     & 0.086    \\ \hline
        LGCN PAS (2.0) & 0.1526                                                    & 0.07573 & 0.1581     & 0.06509  \\ \hline
        NGCF PAS (0.0) & 0.1758                                                    & 0.08483 & 0.2737     & 0.1111   \\ \hline
        NGCF PAS (0.5) & 0.1756                                                    & 0.08472 & 0.2744     & 0.1108   \\ \hline
        NGCF PAS (1.0) & 0.1749                                                    & 0.08485 & 0.2743     & 0.1111   \\ \hline
        NGCF PAS (2.0) & 0.1741                                                    & 0.08371 & 0.2762     & 0.1116   \\ \hline
    \end{tabular}
    \caption{Results for the experiment using different input values.}
    \label{tab:hyperparameter-results}
\end{table*}
As can be seen on \autoref{tab:hyperparameter-results} inserting categories and price into the adjacency matrix decreases the performance compared to LightGCN.

> The value it performs best with is 0.5, followed by 0.0.

Reviewer 2: ```suggestion
The value it has the best performance with is 0.5, followed by 0.0.
```

Giving prices and categories the same or higher input value as the connection between item and user therefore decreases the results.
There could be multiple reasons for this generally not performing well.
The embeddings created from this input could be used, so that it tries to recommend prices and categories, which is not possible.
Another reason could be because LightGCN does not utilize feature transformation and the nonlinear activation function.
LightGCN performs well, if there are only user and item ids, and the decrease in performance could be because LightGCN is unable to handle the semantic representations in price and category.\\
For NGCF changing the input value of price and categories changes almost nothing.
This is likely because the embeddings for price and categories are never used.

> However, we did have an initial idea, that the convolutions with price and category would be reflected onto users and items, which is seen to give a very minimal effect on the outcome.

Reviewer 2: er ikke sikker på hvad du mener med det her ? er det er forhold til ngcf eller lightGCN eller begge ?
